caffe2/python/optimizer.py

```diff
@@ -17,6 +17,15 @@
 
 AuxOptimizerParams = namedtuple("AuxOptimizerParams", ["local", "shared"])
 
+def _get_lr_blob_name():
```

Comment:
Why there will be perf problem? It should create only one learning rate blob per optimization class, so if we did : 
```
def _get_ler_blob_name(optimizer_prefix):
     s = scope.CurrentDeviceScope()
     if s.device_type == caffe2_pb2.CUDA:
         return "%s_lr_gpu_%d" % (optimizer_prefix, s.cuda_gpu_id)
``` 
and called `_get_ler_blob_name(self.__class__.name__)` for every optimizer class.
In your example there will be one lr blob per gpu SgdOptimizer_lr_gpu_0, SgdOptimizer_lr_gpu_0, ..etc